src/pymmcore_widgets/_channel_table_widget.py

```diff
@@ -0,0 +1,228 @@
+        )
+
+        layout.addWidget(self.add_ch_button)
+        layout.addWidget(self.remove_ch_button)
+        layout.addWidget(self.clear_ch_button)
+        layout.addItem(spacer)
+
+        group_layout.addWidget(wdg, 0, 1)
+
+    def _add_channel(self) -> bool:
+        """Add, remove or clear channel table.  Return True if anyting was changed."""
+        if len(self._mmc.getLoadedDevices()) <= 1:
+            return False
+
+        channel_group = self._mmc.getChannelGroup()
+        if not channel_group:
+            warnings.warn("First select Micro-Manager 'ChannelGroup'.")
+            return False
+
+        channel_combobox = self._create_channel_combobox()
+        if not channel_combobox:
+            return False
+
+        channel_exp_spinbox = self._create_exposure_doublespinbox()
+
+        self._add_widgets_to_table(channel_combobox, channel_exp_spinbox)
+
+        channel_combobox.setCurrentIndex(self._get_new_channel_index())
+
+        self.valueChanged.emit()
+
+        return True
+
+    def _create_channel_combobox(self, channel_group: str = "") -> QComboBox | None:
+        channel_combobox = QComboBox()
+
+        ch_group = channel_group or self._mmc.getChannelGroup()
+        if not ch_group:
+            return None
+
+        channel_list = list(self._mmc.getAvailableConfigs(ch_group))
+        channel_combobox.addItems(channel_list)
+        return channel_combobox
+
+    def _create_exposure_doublespinbox(self) -> QDoubleSpinBox:
+        channel_exp_spinbox = QDoubleSpinBox()
+        channel_exp_spinbox.setRange(0, 10000)
+        channel_exp_spinbox.setValue(100)
+        channel_exp_spinbox.setAlignment(AlignCenter)
+        channel_exp_spinbox.valueChanged.connect(self.valueChanged)
+        return channel_exp_spinbox
+
+    def _add_widgets_to_table(
+        self, channel_combo: QComboBox, exp_dspinbox: QDoubleSpinBox
+    ) -> None:
+        idx = self.channel_tableWidget.rowCount()
+        self.channel_tableWidget.insertRow(idx)
+        self.channel_tableWidget.setCellWidget(idx, 0, channel_combo)
+        self.channel_tableWidget.setCellWidget(idx, 1, exp_dspinbox)
+
+    def _get_new_channel_index(self) -> int:
+        if self.channel_tableWidget.rowCount() == 1:
+            return 0
+        idxs = list(range(self.channel_tableWidget.cellWidget(0, 0).count()))
+        used_idxs = []
+        for row in range(self.channel_tableWidget.rowCount() - 1):
+            combo = cast(QComboBox, self.channel_tableWidget.cellWidget(row, 0))
+            used_idxs.append(combo.currentIndex())
+        new_idxs = list(set(idxs) - set(used_idxs))
+        return new_idxs[0] if new_idxs else 0
+
+    def _remove_channel(self) -> None:
+        rows = {r.row() for r in self.channel_tableWidget.selectedIndexes()}
+        for idx in sorted(rows, reverse=True):
+            self.channel_tableWidget.removeRow(idx)
+        self.valueChanged.emit()
```

Comment:
all of these `valueChanged.emit` emissions rely on someone actually clicking a button, but that leaves the possibility of inconsistent states if something programmatically adjust the table.

Note also that the `_create_gui` method in *both* the MDA widget and Sample widget guis have these lines:

```python
self.channel_groupbox.channel_tableWidget.model().rowsInserted.connect(
    self._enable_run_btn
)
self.channel_groupbox.channel_tableWidget.model().rowsRemoved.connect(
    self._enable_run_btn
)
```

So, that needs work.  The `ChannelTable.valueWidget` should emit pretty much any time anything happens that would make the `value()` method return a different value (i.e. anytime the the table changes at all).  And other widgets should connect to _that_ signal, rather than accessing the inner table widget.  That is, it should be `self.channel_groupbox.valueChanged.connect...` rather than `self.channel_groupbox.channel_tableWidget.model().rowsInserted.connect...`